core/src/main/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -17,53 +17,291 @@
 package org.apache.spark.sql.delta.perf
 
 import org.apache.spark.sql.catalyst.InternalRow
-import org.apache.spark.sql.catalyst.expressions.{Alias, Literal}
-import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression, Complete, Count}
+import org.apache.spark.sql.catalyst.expressions.{Alias, AttributeReference, Literal}
+import org.apache.spark.sql.catalyst.expressions.aggregate._
 import org.apache.spark.sql.catalyst.planning.PhysicalOperation
 import org.apache.spark.sql.catalyst.plans.logical._
-import org.apache.spark.sql.delta.DeltaTable
+import org.apache.spark.sql.catalyst.util.{CaseInsensitiveMap, DateTimeUtils}
+import org.apache.spark.sql.delta.{DeltaColumnMapping, DeltaTable, Snapshot}
+import org.apache.spark.sql.delta.commands.DeletionVectorUtils.isTableDVFree
 import org.apache.spark.sql.delta.files.TahoeLogFileIndex
 import org.apache.spark.sql.delta.stats.DeltaScanGenerator
-import org.apache.spark.sql.functions.{coalesce, col, count, lit, sum, when}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+
+import java.sql.Date
+import scala.collection.immutable.HashSet
 
 trait OptimizeMetadataOnlyDeltaQuery {
   def optimizeQueryWithMetadata(plan: LogicalPlan): LogicalPlan = {
     plan.transformUpWithSubqueries {
-      case agg@CountStarDeltaTable(countValue) =>
-        LocalRelation(agg.output, Seq(InternalRow(countValue)))
+      case agg@AggregateDeltaTable(tahoeLogFileIndex) =>
```

Comment:
Probably not. Show() adds CAST to the columns. I need to find a clean solution to handle it, this new implementation is more complex because it doesn't assume a specific query format like previous one. We can have SELECT COUNT(*), CAST(COUNT(*) AS String) FROM TABLE.

Reply:
I added a special case for COUNT + show(). I hope this helps for now.

Reply:
Implemented MIN/MAX in show command. I restricted it for Integral types only, for safety

Reply:
Changed to support Spark 3.5 show(), which uses ToPrettyString instead of CAST